index.js

```diff
@@ -137,4 +137,50 @@ newSightingForm.addEventListener('submit', function (event) {
 /***** End of Starter Code *****/
 /************************** EVENTS PART 2 JS MINI CHALLENGE ******************************/
 
+//Deliverable 1: When a user clicks an animal sighting's like button, that animal sighting's likes should increase by 1
+const animalCollection = document.querySelector('ul#animals')
+
+animalCollection.addEventListener('click', function(event){
+    // console.log(event.target)
+
+    if (event.target.matches('button.like-button')){
+        // event.target.className === 'like-button' another option
+        console.log(event.target)
+        const li = event.target.closest('li') //li element of animal sighting
+        const likesPtag = li.querySelector('p.likes-display') //get p tag of likes in that li element
+        const currLikes = parseInt(likesPtag.textContent)  //get likes as an integer
+        likesPtag.textContent = `${currLikes + 1} Likes` //add a like 
+    }
+//Deliverable 2: When a user clicks an animal sighting's delete button, that animal sighting should be removed from the page.
+    else if(event.target.className === 'delete-button'){ //can also use matches
+        // console.log('delete clicked!')
+        const li = event.target.closest('li')
+        li.remove()
+    }
+//Deliverable 3: When a user clicks an animal sighting's 'Toggle Update Form' button, the form associated with that animal sighting should toggle between displaying and not displaying on the page.
+    else if(event.target.matches('button.toggle-update-form-button')){
+        console.log('clicked')
+        const form = li.querySelector('form') //find the form
+        if(form.style.display === 'block'){
+            form.style.display = 'none'
+        }
+        else{
+            form.style.display = 'block'
+        }
+        //form.style.display = form.style.display === 'block' ? 'none' : 'block'
+    }
+})
```

Comment:
Nice! You have an event listener on the entire animals UL element, and based on the class names you are delegating out different logic. Great. First step is getting everything up and running. Awesome. Next, we can refactor for readability and to make our code more reusable.

```
/***** DELIVERABLE 1, 2, & 3 part 1 *****/

const animalsUL = document.querySelector('ul#animals')

animalsUL.addEventListener("click", function(event){
    if (event.target.matches('button.like-button')){
        handleLike(event)
    } else if (event.target.matches('button.delete-button')){
        handleDelete(event)
    } else if (event.target.matches('button.toggle-update-form-button')){
        handleToggleUpdate(event)
    }
})

/***** DELIVERABLE 1 part 2 *****/
function handleLike(e){
    const likesDisplayPTag = e.target.previousElementSibling
    const currentTextContent = likesDisplayPTag.innerText
    let numberOfLikes = parseInt(currentTextContent.split(" ")[0])
    numberOfLikes += 1
    const newTextContent = `${numberOfLikes} Likes`
    likesDisplayPTag.innerText = newTextContent
}

/***** DELIVERABLE 2 part 2 *****/
function handleDelete(e){
    const parentLI = e.target.parentElement
    parentLI.remove()
}

/***** DELIVERABLE 3 part 2 *****/
function handleToggleUpdate(e){
    const updateForm = e.target.nextElementSibling
    toggleHidden(updateForm)
}

/***** DELIVERABLE 3 part 3 *****/
function toggleHidden(element){
    element.style.display === "none" ? element.style.display = "block" : element.style.display = "none"
}
```